### PR TITLE
Greek/Character Class files

### PR DESCRIPTION
**Pull request by iamsitting**
Module to upload and view class files.

### CODE REVIEW

**Comment by iamsitting** [comment]
@yipingbenjaminliu this should be ready for front-end magic. 🤞

**Comment by yipingbenjaminliu** [comment]
only thing left would be to redirect trainee to the right page and bypassing upload view, but that should take stuff from the rolls or schedules model, other than that, we're good for this module

File: ap/classes/models.py [hunk 1]
```diff
@@ -34,3 +40,26 @@ def save(self, *args, **kwargs):
     super(Class, self).save(*args, **kwargs)
 
   objects = ClassManager()
+
+
+file_storage = FileSystemStorage(
```
Comment (jontey) [inline ap/classes/models.py]: I don't think you need this. FileField should use the MEDIA_ROOT that is already defined in settings by default

File: ap/classes/urls.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from django.conf.urls import url
+
+from classes import views
+
+urlpatterns = [
+  url(r'^$', views.upload, name='index'),
```
Comment (jontey) [inline ap/classes/urls.py]: This redirects everyone to the upload page, which only administrators can view. This should not be the default view for everyone though

File: ap/classes/views.py [hunk 3]
```diff
@@ -1 +1,39 @@
...
+
+def class_files(request, classname):
+  if request.method == 'GET':
+    ctx = {}
+
+    if classname is None:
+      ctx['class_files'] = ClassFile.objects.all()
+      ctx['page_title'] = 'Class Files'
+      ctx['form'] = ClassFileForm()
+      ctx['delete'] = True
```
Comment (jontey) [inline ap/classes/views.py]: Change delete to True only if user has group=administrator

File: ap/classes/templates/classes/upload.html [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+<div class="navbar-header">
+<button type="button" class="navbar-toggle glyphicon glyphicon-chevron-down visible-xs" data-toggle="collapse" data-target="#navbar-collapse-2">
+	</button>
+	<a class="navbar-brand" href="{% url 'classes:index' %}">{{page_title}}</a>
+</div>
+
+<div class="collapse navbar-collapse" role="tablist" id="navbar-collapse-2">
+	<ul class="nav nav-tabs" >
+
+		<li {% if current_url == 'Greek' %}class=active{% endif %}><a href="{% url 'classes:classfiles' 'Greek' %}">Greek</a></li>
```
Comment (jontey) [inline ap/classes/templates/classes/upload.html]: Change to 2 spaces tabs for this file

File: ap/classes/templates/classes/file_list.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  <div>
+    <table class="table">
+      <thead>
+      <tr>
+        <th>File</th>
+        <th>Class</th>
+        {% if delete %}
+        <th>Delete</th>
+        {% endif %}
+      </tr>
```
Comment (jontey) [inline ap/classes/templates/classes/file_list.html]: Delete trailing spaces

File: ap/classes/models.py [hunk 6]
```diff
@@ -34,3 +40,26 @@ def save(self, *args, **kwargs):
...
+
+  CLASS_CHOICES = (
+    ('Greek', 'Greek'),
+    ('German', 'German'),
+    ('Character', 'Character')
+  )
+
+  for_class = models.CharField(max_length=10, choices=CLASS_CHOICES, null=True)
+
+  file = models.FileField(upload_to='class_files', storage=file_storage)
```
Comment (jontey) [inline ap/classes/models.py]: Is it possible to change upload_to to save to MEDIA_ROOT/class_files/[for_class]/ ?
That way you don't have a mix of files for all classes in a single folder

File: ap/classes/admin.py [hunk 7]
```diff
@@ -18,8 +18,9 @@ class Meta:
     model = Class
     exclude = []
     widgets = {
-    'schedules': admin.widgets.FilteredSelectMultiple(
-      "schedules", is_stacked=False),
+      'schedules': admin.widgets.FilteredSelectMultiple(
```
Comment (jontey) [inline ap/classes/admin.py]: Is there a reason you added this? It is also declared on line 14

File: ap/classes/admin.py [hunk 8]
```diff
@@ -43,5 +44,6 @@ def save_model(self, request, obj, form, change):
 #         obj.type = 'C'
```
Comment (jontey) [inline ap/classes/admin.py]: Indentation is off here

**Pull request merged**